src/account/UpgradeableModularAccount.sol

```diff
@@ -302,6 +304,31 @@ contract UpgradeableModularAccount is
         _uninstallPlugin(plugin, manifest, pluginUninstallData, hookUnapplyData);
     }
 
+    // Should we check versions here?
+    function replacePlugin(
+        address oldPlugin,
+        address newPlugin,
+        bytes32 manifestHash,
+        bytes[] calldata hookUnapplyData
+    ) external override wrapNativeFunction {
+        address oldPluginRegistry = BasePlugin(oldPlugin).getVersionRegistry();
+        address newPluginRegistry = BasePlugin(newPlugin).getVersionRegistry();
+
+        require(oldPluginRegistry == newPluginRegistry, "Plugins use different VersionRegistries");
```

Comment:
To maintain the code consistency, I think it would be better to use if - revert statement rather than require.